docker-storage-setup.sh

```diff
@@ -931,49 +932,88 @@ create_lv() {
-    Fatal "Specify a valid value for DOCKER_ROOT_VOLUME_SIZE."
+setup_extra_volume() {
+  if [ -z "$3" ]; then
+    Fatal "Specify a valid value for EXTRA_VOLUME_SIZE."
   fi
 
-  if ! check_data_size_syntax $DOCKER_ROOT_VOLUME_SIZE; then
-    Fatal "DOCKER_ROOT_VOLUME_SIZE value $DOCKER_ROOT_VOLUME_SIZE is invalid."
+  if ! check_data_size_syntax $3; then
+    Fatal "EXTRA_VOLUME_SIZE value $3 is invalid."
   fi
 
-  if ! create_lv $DOCKER_ROOT_VOLUME_SIZE $DOCKER_ROOT_LV_NAME; then
-    Fatal "Failed to create volume $DOCKER_ROOT_VOLUME_SIZE of size ${DOCKER_ROOT_VOLUME_SIZE}."
+  if ! create_lv $3 $1; then
+    Fatal "Failed to create volume $1 of size ${3}."
   fi
 
-  if ! mkfs -t xfs /dev/$VG/$DOCKER_ROOT_LV_NAME > /dev/null; then
-    Fatal "Failed to create filesystem on /dev/$VG/${DOCKER_ROOT_LV_NAME}."
+  if ! mkfs -t xfs /dev/$VG/$1 > /dev/null; then
+    Fatal "Failed to create filesystem on /dev/$VG/${1}."
   fi
 
-  if ! mount_docker_root_volume; then
-    Fatal "Failed to mount docker root volume ${DOCKER_ROOT_LV_NAME} on ${DOCKER_ROOT_DIR}"
+  if ! mount_extra_volume $1 $2; then
+    Fatal "Failed to mount volume ${1} on ${2}"
   fi
 
   # setup right selinux label first time fs is created. Mount operation
   # changes the label of directory to reflect the label on root inode
   # of mounted fs.
-  if ! restore_selinux_context $DOCKER_ROOT_DIR; then
+  if ! restore_selinux_context $2; then
     return 1
   fi
 }
 
+setup_extra_lv_fs() {
+  [ -z "$EXTRA_VOLUME_MOUNT_DIR" ] && [ -z "$EXTRA_VOLUME_NAME" ] && return 0
+  local resolved_mount_dir_path=$RESOLVED_MOUNT_DIR_PATH
+  local extra_volume_name=$EXTRA_VOLUME_NAME
+  local extra_volume_size=$EXTRA_VOLUME_SIZE
+
+  if ! setup_extra_dir $resolved_mount_dir_path; then
+    return 1
+  fi
+  if extra_volume_exists $extra_volume_name; then
+    return 0
+  fi
+  # Container runtime extra volume does not exist. Create one.
+  if ! setup_extra_volume $extra_volume_name $resolved_mount_dir_path $extra_volume_size; then
+    Fatal "Failed to setup logical volume for container root."
+  fi
+}
+
 setup_docker_root_lv_fs() {
   [ "$DOCKER_ROOT_VOLUME" != "yes" ] && return 0
   if ! setup_docker_root_dir; then
     return 1
   fi
-  if docker_root_volume_exists; then
+  local docker_root_dir=$DOCKER_ROOT_DIR
+  local docker_root_lv_name=$DOCKER_ROOT_LV_NAME
+  local docker_root_volume_size=$DOCKER_ROOT_VOLUME_SIZE
+
+  if extra_volume_exists $docker_root_lv_name; then
     return 0
   fi
   # Docker root volume does not exist. Create one.
-  if ! setup_docker_root_volume; then
+  if ! setup_extra_volume $docker_root_lv_name $docker_root_dir $docker_root_volume_size; then
     Fatal "Failed to setup logical volume for docker root."
   fi
 }
 
+check_storage_options(){
+  if [ "$DOCKER_ROOT_VOLUME" == "yes" ] && [ -n "$EXTRA_VOLUME_MOUNT_DIR" ];then
+     Fatal "DOCKER_ROOT_VOLUME and EXTRA_VOLUME_MOUNT_DIR are mutually exclusive options. Use either one of them, not both"
```

Comment:
Lets get rid of `Use either one of them, not both`. Once we say "mutually exclusive", it is implied.